typescript/src/templater.ts

```diff
@@ -55,12 +56,133 @@ async function recursiveFileList(
   return result;
 }
 
+function createSnippetIndexMetadata(
+  api: API,
+  basePath: string
+): protos.google.cloud.tools.snippetgen.snippetindex.v1.IIndex {
+  const apis: protos.google.cloud.tools.snippetgen.snippetindex.v1.IApi[] = [];
+
+  const clientLibrary: protos.google.cloud.tools.snippetgen.snippetindex.v1.IClientLibrary = {
+    name: `nodejs-${api.naming.productName.toKebabCase()}`,
+    version: '0.1.0',
+    language: ('TYPESCRIPT' as unknown) as protos.google.cloud.tools.snippetgen.snippetindex.v1.Language,
+    apis: [{id: api.naming.protoPackage, version: api.naming.version}],
+  };
+
+  const snippets = createSnippetMetadata(api, basePath);
+  return {clientLibrary, snippets};
+}
+
+function createSnippetMetadata(
```

Comment:
Idea for future refactor, it might be nice to pull the logic for generating the snippet index and individual snippet metadata into its own helper file.